netbout-web/src/test/java/com/netbout/rest/MarkdownTxtmarkTest.java

```diff
@@ -0,0 +1,274 @@
+
+/**
+ * Test case for {@link MarkdownTxtmark}.
+ *
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 2.23
+ * @todo #867:30min/DEV MarkdownTxtmark not implemented but has to be.
+ *  See #847 for details.
+ *  This class should be implementation of Markdown using TxtMark.
+ *  See https://github.com/rjeschke/txtmark.
+ *  After that don't forget remove Ignore annotation from unit tests.
+ *
+ */
+public final class MarkdownTxtmarkTest {
+    /**
+     * MarkdownTxtmark can handle whitespace after links.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    @Ignore
+    public void handlesWhitespaceAfterLinks() throws Exception {
+        MatcherAssert.assertThat(
+            new MarkdownTxtmark().html(
+                "Hi [google](http://www.google.com) how are you?"
+            ),
+            Matchers.equalTo(
+                MarkdownTxtmarkTest.replaceNewLine(
+                    // @checkstyle LineLengthCheck (1 line)
+                    "<p>Hi \n<a href=\"http://www.google.com\">google</a> how are you?</p>\n"
+                )
+            )
+        );
+    }
+
+    /**
+     * MarkdownTxtmark can format a text to HTML.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    @Ignore
+    public void formatsTextToHtml() throws Exception {
+        final String meta = new MarkdownTxtmark().html(
+            "**hi**, _dude_!\r\n\n     b**o\n    \n    \n    o**m\n"
+        );
+        MatcherAssert.assertThat(
+            String.format("<x>%s</x>", meta),
+            Matchers.describedAs(
+                meta,
+                XhtmlMatchers.hasXPaths(
+                    "/x/p/strong[.='hi']",
+                    "/x/p/em[.='dude']",
+                    "/x/pre/code[.=' b**o\n\n\no**m\n']"
+                )
+            )
+        );
+    }
+
+    /**
+     * MarkdownTxtmark can handle broken formatting correctly.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    @Ignore
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public void handlesBrokenFormattingGracefully() throws Exception {
+        final String[] texts = {
+            "**\n ",
+            "__",
+            "",
+            "**hi there! {{{",
+            "    \n  \n      \n     \n",
+        };
+        for (final String text : texts) {
+            MatcherAssert.assertThat(
+                String.format("<z>%s</z>", new MarkdownTxtmark().html(text)),
+                XhtmlMatchers.hasXPath("/z")
+            );
+        }
+    }
+
+    /**
+     * MarkdownTxtmark can format small snippets.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    @Ignore
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public void formatsTextFragmentsToHtml() throws Exception {
+        final String[][] texts = {
+            new String[] {"hi, *dude*!", "<p>hi, \n<em>dude</em>!</p>"},
```

Comment:
@dmzaytsev same here and in all other places, where you are using new line symbols.